components/wpt-results.html

```diff
@@ -46,59 +120,192 @@
 
       static get properties() {
         return {
+          query: {
+            type: String,
+            value: '',
+            observer: '_queryChanged'
+          },
           specs: {
             type: Array,
-            observer: '_specsChanged',
             value: []
-          }
+          },
+          testRuns: {
+            type: Array,
+          },
         };
       }
 
-      connectedCallback() {
+      async connectedCallback() {
         super.connectedCallback();
 
-        this._fetchResults()
+        const testFileResults = await Promise.all(this.testRuns.map(testRun => {
+          return this._fetchResults(testRun.results_url)
+        }))
+
+        let specs = []
+
+        testFileResults.forEach(result => {
+          const testFiles = result.testFiles
+          const resultsURL = result.resultsURL
+
+          Object.keys(result.testFiles).forEach(testFileName => {
+            const specName = this._specFromTestPath(testFileName)
+
+            let spec = this._arrayElemByProperty(specs, 'specName', specName, {
+              results: [], specName: specName, testFiles: []
+            })
```

Comment:
This section contains a lot of gymnastics to work with arrays of objects since `dom-repeat` only works with arrays. It would be much easier to simply deal with objects. The next step I want to take is to write my own version of `dom-repeat` that takes an object and supports easier nested repeating to better fit our use case.